includes/wikia/Wikia.php

```diff
@@ -584,54 +584,23 @@ static public function debugBacktrace($method) {
 	 * @access public
 	 * @static
 	 *
-	 * @param String $lang  -- language code
+	 * @param String $langCode  -- language code
 	 *
 	 * @return User -- instance of user object
 	 */
 	static public function staffForLang( $langCode ) {
-		wfProfileIn( __METHOD__ );
-
-		$staffSigs = wfMsgExt('staffsigs', array('language'=>'en')); // fzy, rt#32053
-
-		$staffUser = false;
-		if( !empty( $staffSigs ) ) {
-			$lines = explode("\n", $staffSigs);
-
-			$data = array();
-			$sectLangCode = '';
-			foreach ( $lines as $line ) {
-				if( strpos( $line, '* ' ) === 0 ) {
-					//language line
-					$sectLangCode = trim( $line, '* ' );
-					continue;
-				}
-				if( strpos( $line, '* ' ) == 1 && $sectLangCode ) {
-					//user line
-					$user = trim( $line, '** ' );
-					$data[$sectLangCode][] = $user;
-				}
-			}
+		global $wgFounderWelcomeAuthor;
 
-			//did we get any names for our target language?
-			if( !empty( $data[$langCode] ) ) {
-				//pick one
-				$key = array_rand($data[$langCode]);
-
-				//use it
-				$staffUser = User::newFromName( $data[$langCode][$key] );
-				$staffUser->load();
-			}
+		if ( array_key_exists( $langCode, $wgFounderWelcomeAuthor ) ) {
+			$key = array_rand( $wgFounderWelcomeAuthor[$langCode] );
+			$staffUser = User::newFromName( $wgFounderWelcomeAuthor[$langCode][$key] );
+		} else {
+			// Fallback to robot when there is no explicit welcoming user set (unsupported language)
+			$staffUser = User::newFromName( 'Fandom' );
```

Comment:
Good question.

Pre-rebranding, we had `Wikia` for welcomes etc. and `WikiaBot` for menial labor. They have been rebranded `Fandom` and `FandomBot`, respectively (`Wikia::BOT_USER` points to the latter). I believe we should continue to use "Fandom" for consistency here.